shared/src/main/scala/async/futures.scala

```diff
@@ -150,11 +150,10 @@ object Future:
 
   end RunnableFuture
 
-  /** Create a future that asynchronously executes `body` that defines its result value in a Try or returns failure if
-    * an exception was thrown. If the future is created in an Async context, it is added to the children of that
-    * context's root.
+  /** Create a future that asynchronously executes [[body]] that defines its result value in a [[Try]] or returns
+    * [[Failure]] if an exception was thrown.
     */
-  def apply[T](body: Async ?=> T)(using Async): Future[T] =
+  def apply[T](body: Async.Spawnable ?=> T)(using async: Async, spawnable: Async.Spawnable & async.type): Future[T] =
```

Comment:
Is the Scala Language specification actually explicit enough that this is guaranteed to behave as we want (i.e., in a scenario where `Spawnable { NonSpawnable { Future.apply(...) } }`, the code does not compile with the current typing while it does when just `using async: Async.Spawnable`)?

Reply:
I think so, I'm pretty sure `using async: Async` will resolve to the closest `Async` instance (from `NonSpawnable`), which will cause the `using spawnable: Spawnable & async.type` search to fail. There should be no backtracking behavior. 
Of course if you manually specify the `Async` instance from `Spawnable { }` it will compile but that's on you for doing that

Reply:
Okay, great!